src/pe/data_directories.rs

```diff
@@ -23,6 +117,22 @@ pub struct DataDirectories {
     pub data_directories: [Option<DataDirectory>; NUM_DATA_DIRECTORIES],
 }
 
+impl ctx::TryIntoCtx<scroll::Endian> for DataDirectories {
+    type Error = error::Error;
+
+    fn try_into_ctx(self, bytes: &mut [u8], ctx: scroll::Endian) -> Result<usize, Self::Error> {
+        let offset = &mut 0;
+        for opt_dd in self.data_directories {
+            if let Some(dd) = opt_dd {
+                bytes.gwrite_with(dd.inner, offset, ctx)?;
+            } else {
+                bytes.gwrite(&[0; SIZEOF_DATA_DIRECTORY][..], offset)?;
+            }
+        }
+        Ok(NUM_DATA_DIRECTORIES * SIZEOF_DATA_DIRECTORY)
```

Comment:
I was expecting the offset returned here, but I assume this is the mandatory size for data directories always?

Reply:
We honestly can shuffle things around and put the offset here

It's actually not the mandatory size for data directories I discovered recently.
You can have a smaller size, but the idea is that if you have data directory 4, 5, 6. You need to allocate for the first 6 directories.

If you have 11, 12. The first 12.

etc.

You always need to allocate for the first max(index of a valid data directory) basically from what I gathered in the specification.

Reply:
(because if you allocate less, you have no way to understand that it's the 11th that is valid and not the 1st for example, but you can understand that if you allocated less than N data directories, the N + 1th, N + 2th are None.)

Reply:
I reintroduced the offset in this parsing function.

Reply:
Resolving, let me know if still relevant.